src/features/game/lib/gameMachine.ts

```diff
@@ -200,17 +208,14 @@ export function startGame(authContext: Options) {
               return { state: INITIAL_FARM, onChain };
             },
             onDone: [
+              {
+                target: "announcing",
+                cond: () => hasAnnouncements(),
+                actions: "assignGame",
+              },
               {
                 target: "playing",
-                actions: assign({
-                  state: (_, event) => event.data.state,
-                  onChain: (_, event) => event.data.onChain,
-                  owner: (_, event) => event.data.owner,
-                  offset: (_, event) => event.data.offset,
-                  sessionId: (_, event) => event.data.sessionId,
-                  fingerprint: (_, event) => event.data.fingerprint,
-                  itemsMintedAt: (_, event) => event.data.itemsMintedAt,
-                }),
+                actions: "assignGame",
```

Comment:
One note about this refactor that is important to mention is that we lose type safety when we move actions down to the options object.

You can fix this by inferring the event type below.